crates/ruff_server/src/lint.rs

```diff
@@ -0,0 +1,119 @@
+//! Access to the Ruff linting API for the LSP
+
+use std::path::Path;
+
+use ruff_diagnostics::{Applicability, Diagnostic, DiagnosticKind, Fix};
+use ruff_linter::{
+    directives::{extract_directives, Flags},
+    linter::{check_path, LinterResult, TokenSource},
+    registry::AsRule,
+    settings::{flags, LinterSettings},
+    source_kind::SourceKind,
+};
+use ruff_python_ast::PySourceType;
+use ruff_python_codegen::Stylist;
+use ruff_python_index::Indexer;
+use ruff_python_parser::lexer::LexResult;
+use ruff_python_parser::AsMode;
+use ruff_source_file::Locator;
+use serde::{Deserialize, Serialize};
+
+use crate::{PositionEncoding, DIAGNOSTIC_NAME};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct DiagnosticFix {
+    pub(crate) kind: DiagnosticKind,
+    pub(crate) fix: Fix,
+}
+
+pub(crate) fn check(
+    document: &crate::edit::Document,
+    linter_settings: &LinterSettings,
+    encoding: PositionEncoding,
+) -> Vec<lsp_types::Diagnostic> {
```

Comment:
It would be nice if we could reuse more of the `ruff_linter` crate methods but maybe something to consider to do later.